### PR TITLE
[16.0][IMP] base_tier_validation*: update test mechanism to support testing of messages

### PR DESCRIPTION
**Pull request by bosd**

### CODE REVIEW

**Comment by bosd** [comment]
The use sudo commit  is needed to fix:

```

ERROR: TierTierValidation.test_01_reviewer_from_python_expression
Traceback (most recent call last):
  File "/__w/server-ux/server-ux/base_tier_validation_formula/tests/test_tier_validation.py", line 92, in test_01_reviewer_from_python_expression
    reviews = self.test_record.with_user(self.test_user_3.id).request_validation()
  File "/__w/server-ux/server-ux/base_tier_validation/models/tier_validation.py", line 474, in request_validation
    self._notify_review_requested(created_trs)
  File "/__w/server-ux/server-ux/base_tier_validation/models/tier_validation.py", line 446, in _notify_review_requested
    getattr(rec, post)(
  File "/opt/odoo/addons/mail/models/mail_thread.py", line 2005, in message_post
    author_id, email_from = self._message_compute_author(author_id, email_from, raise_on_email=True)
  File "/opt/odoo/addons/mail/models/mail_thread.py", line 2290, in _message_compute_author
    raise exceptions.UserError(_("Unable to send message, please configure the sender's email address."))
odoo.exceptions.UserError: Unable to send message, please configure the sender's email address.

```

**Comment by bosd** [comment]
base_tier_validation_forward patch fixes:
<details>

```

ERROR: TierTierValidation.test_01_forward_tier
Traceback (most recent call last):
  File "/__w/server-ux/server-ux/base_tier_validation_forward/tests/test_tier_validation.py", line 110, in test_01_forward_tier
    wiz.add_forward()
  File "/__w/server-ux/server-ux/base_tier_validation_forward/wizard/forward_wizard.py", line 34, in add_forward
    prev_reviews = prev_comment.add_comment()
  File "/__w/server-ux/server-ux/base_tier_validation_forward/wizard/comment_wizard.py", line 13, in add_comment
    rec._forward_tier(self.review_ids)
  File "/__w/server-ux/server-ux/base_tier_validation_forward/models/tier_validation.py", line 72, in _forward_tier
    rec._notify_forwarded_reviews()
  File "/__w/server-ux/server-ux/base_tier_validation_forward/models/tier_validation.py", line 78, in _notify_forwarded_reviews
    getattr(self, post)(
  File "/opt/odoo/addons/mail/models/mail_thread.py", line 2005, in message_post
    author_id, email_from = self._message_compute_author(author_id, email_from, raise_on_email=True)
  File "/opt/odoo/addons/mail/models/mail_thread.py", line 2290, in _message_compute_author
    raise exceptions.UserError(_("Unable to send message, please configure the sender's email address."))
odoo.exceptions.UserError: Unable to send message, please configure the sender's email address.
 
```
</details>

**Comment by bosd** [comment]
refactored using `DISABLED_MAIL_CONTEXT`
as mentioned in #739

**Review by Dranyel-Bosd — APPROVED** [review]
functionally review

File: base_tier_validation/models/tier_validation.py [hunk 1]
```diff
@@ -435,7 +435,7 @@ def _notify_review_requested(self, tier_reviews):
         subscribe = "message_subscribe"
         post = "message_post"
         if hasattr(self, post) and hasattr(self, subscribe):
-            for rec in self:
+            for rec in self.sudo():
```
Comment (LoisRForgeFlow) [inline base_tier_validation/models/tier_validation.py:438]: why?

Comment (LoisRForgeFlow) [reply base_tier_validation/models/tier_validation.py:438]: Oh, You already posted that info here https://github.com/OCA/server-ux/pull/747#issuecomment-1793072646.

Comment (LoisRForgeFlow) [reply base_tier_validation/models/tier_validation.py:438]: I am not sure that fixing the tests is enough reason to add that sudo... could you find another solution?

Comment (bosd) [reply base_tier_validation/models/tier_validation.py:438]: Well, all other message posts have been refactored to use sudo.
This one is the left over.. 

Idk how to solve differently. Any suggestions?

Comment (LoisRForgeFlow) [reply base_tier_validation/models/tier_validation.py:438]: I see, thanks for pointing that out

**Review by LoisRForgeFlow — APPROVED** [review]
/ocabot merge patch

**Comment by bosd** [comment]
Hmm.. Tests on this one passed.. While it should have failed.
After the merge of this pr. CI is broken as discovered in #748 

image

cc: @simohawk

**Pull request merged**